tools/cldr-code/src/main/java/org/unicode/cldr/test/ExampleGenerator.java

```diff
@@ -2411,6 +2418,36 @@ private String handleTimeZoneName(XPathParts parts, String value) {
         return result;
     }
 
+    private String generateAltPattern(String dateTimeValue) {
+        fp.set(dateTimeValue);
+            StringBuilder altPatternBuilder = new StringBuilder();
+            for (Object item : fp.getItems()) {
+                if (item instanceof DateTimePatternGenerator.VariableField) {
+                    DateTimePatternGenerator.VariableField field = (DateTimePatternGenerator.VariableField) item;
+                    switch (field.toString()) {
+                        case "h":
+                            altPatternBuilder.append("hh");
+                            break;
+                        case "H":
+                            altPatternBuilder.append("HH");
+                            break;
+                        case "hh":
+                            altPatternBuilder.append("h");
+                            break;
+                        case "HH":
+                            altPatternBuilder.append("H");
+                            break;
+                        default:
+                            altPatternBuilder.append(field.toString());
+                            break;
+                    }
+                } else {
+                    altPatternBuilder.append(item);
```

Comment:
If I understand correctly, if item is not a VariableField, it's a literal string. The ticket description mentions using fp.toString(i, i + 1). Is this acceptable in its place?

Reply:
You want to leave the literals completely alone. So the above looks good.